core/build.gradle.kts

```diff
@@ -85,7 +86,7 @@ dependencies {
   implementation("com.google.code.findbugs:jsr305:3.0.2")
 
   antlr("org.antlr:antlr4:${ANTLR_VERSION}")
-  implementation("org.antlr:antlr4-runtime:${ANTLR_VERSION}")
+  shadow("org.antlr:antlr4-runtime:${ANTLR_VERSION}")
```

Comment:
This doesn't actually remove the ANTLR dependency from the POM file that gets generated for Maven:
```
substrait-java % gw generatePomFileForMaven-publishPublication;grep -i antlr -C1 ./core/build/publications/maven-publish/pom-default.xml 

BUILD SUCCESSFUL in 285ms
2 actionable tasks: 2 executed
    <dependency>
      <groupId>org.antlr</groupId>
      <artifactId>antlr4</artifactId>
      <version>4.13.1</version>
--
    <dependency>
      <groupId>org.antlr</groupId>
      <artifactId>antlr4-runtime</artifactId>
      <version>4.13.1</version>
```
That's why I ended up using the custom configuration - dunno if there's something better: https://github.com/substrait-io/substrait-java/pull/258/files#diff-a6fced4b4e4a99784a2b92028b1adbc8c0ca28ff54e264b0e73727b3d4a460cfR77